platform/scheduler/posix/src/mender-scheduler.c

```diff
@@ -137,24 +126,20 @@ mender_scheduler_init(void) {
     return MENDER_OK;
 }
 
-static mender_err_t
-mender_scheduler_work_delete(mender_scheduler_work_context_t *work_context) {
-    /* Release memory */
-    timer_delete(work_context->timer_handle);
-    pthread_mutex_destroy(&work_context->sem_handle);
-    free(work_context->params.name);
-
-    return MENDER_OK;
-}
-
-static mender_err_t
-mender_scheduler_work_create(mender_scheduler_work_params_t *work_params, mender_scheduler_work_context_t *work_context) {
+mender_err_t
+mender_scheduler_work_create(mender_scheduler_work_params_t *work_params, mender_work_t **work) {
     assert(NULL != work_params);
     assert(NULL != work_params->function);
     assert(NULL != work_params->name);
+    assert(NULL != work);
 
     /* Create work context */
-    memset(work_context, 0, sizeof(mender_scheduler_work_context_t));
+    mender_platform_work_t *work_context = (mender_platform_work_t *)malloc(sizeof(mender_platform_work_t));
+    if (NULL == work_context) {
+        mender_log_error("Unable to allocate memory");
+        goto FAIL;
+    }
+    memset(work_context, 0, sizeof(mender_platform_work_t));
```

Comment:
Can consider using `calloc()` 
```suggestion
    mender_platform_work_t *work_context = (mender_platform_work_t *)calloc(1, sizeof(mender_platform_work_t));
    if (NULL == work_context) {
        mender_log_error("Unable to allocate memory");
        goto FAIL;
    }
```

Reply:
Good point!